### DISCUSSION TITLE
Odd Star Camera Data on 2003-01-29 Using GraceL1b2StarCamera

### DISCUSSION THREAD

**Discussion by ReubenLi1999** [groops-devs/groops#296] [Q&A]
Hi developers,

I have encountered an issue while using [GraceL1b2StarCamera](https://groops-devs.github.io/groops/html/GraceL1b2StarCamera.html) for processing data from 2003-01-29. The star camera data on this date seems unusual (see the figures below).

**Left Panel:** Shows the quaternions for the second arc on 2003-01-28 (this seems correct).
**Right Panel:** Shows the quaternions for the second arc on 2003-01-29 (this seems not correct).
As shown, the curves in the left panel are consistent, while those in the right panel are not.

![image](https://github.com/user-attachments/assets/519df400-2040-4531-ba0f-c37a8a77b8fd)

Additionally, I compared the quaternion observations on 2003-01-29 (blue line in the figure below) with the simulated data generated using [SimulateStarCameraGrace](https://groops-devs.github.io/groops/html/SimulateStarCameraGrace.html) (orange line, with the difference shown in green). In the central portion of the figure, there appears to be a scale factor of -1 between the observation and simulation.

![image](https://github.com/user-attachments/assets/ad90a7b5-73b0-45ed-949c-12fad2c3c556)

Question: I'm uncertain if this discrepancy is due to the GRACE data itself or if it might be a bug in GROOPS. Could you help me understand what might be causing this issue?

Thank you very much for your assistance!

Cheers,
Haosi

**Comment by tmayerguerr** [comment]
Hi Haosi,
Maybe it is not a error/bug, but it is just the definition of quaternions. You can change the sign of all elements without changing the attitude. (q0,qx,qy,qz) and (-q0,-qx,-qy,-qz) represent the same orientation. You could test this with [InstrumentStarCamera2RotaryMatrix](https://groops-devs.github.io/groops/html/InstrumentStarCamera2RotaryMatrix.html).
regards
Torsten

**Reply by ReubenLi1999** [reply]
Hi Torsten,

Thank you for the prompt reply. However, I don’t believe the issue is solely due to the definition of quaternions.

Please see the figure below:

- Left Panel (2003-01-28): The green line (difference) was computed using [InstrumentStarCameraMultiply](https://groops-devs.github.io/groops/html/InstrumentStarCameraMultiply.html) by multiplying the SCA measurement with the inverse of the simulation.
- Right Panel (2003-01-29): Here, the green curve does not remain close to zero as in the left panel.

![image](https://github.com/user-attachments/assets/ae6b5b05-e4b2-40c6-a23e-065ec593c931)

This discrepancy might be due to a temporal pattern difference in the scale factors between the SCA quaternions and simulations. The difference (green line) in the figure below is from the direct subtraction of SCA quaternions and the simulations.

![image](https://github.com/user-attachments/assets/ac18adf9-e48c-4004-a5b2-bab4241b234d)

Thank you for your assistance!

Best regards,
Haosi
